pkg/validation/policy_template.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2025 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"encoding/json"
+	"fmt"
+
+	kyvernoapiv1 "github.com/kyverno/kyverno/api/kyverno/v1"
+
+	kubermaticv1 "k8c.io/kubermatic/sdk/v2/apis/kubermatic/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+var (
+	validPolicyVisibilities = sets.New(kubermaticv1.PolicyTemplateVisibilityGlobal, kubermaticv1.PolicyTemplateVisibilityProject, kubermaticv1.PolicyTemplateVisibilityCluster)
+)
+
+// ValidatePolicyTemplate validates the PolicyTemplate resource.
+func ValidatePolicyTemplate(template *kubermaticv1.PolicyTemplate) field.ErrorList {
+	var allErrs field.ErrorList
+	specPath := field.NewPath("spec")
+
+	if template.Spec.Title == "" {
+		allErrs = append(allErrs, field.Required(specPath.Child("title"), "title is required"))
+	}
+
+	if template.Spec.Description == "" {
+		allErrs = append(allErrs, field.Required(specPath.Child("description"), "description is required"))
+	}
+
+	if !validPolicyVisibilities.Has(template.Spec.Visibility) {
+		allErrs = append(allErrs, field.NotSupported(specPath.Child("visibility"), template.Spec.Visibility, validPolicyVisibilities.UnsortedList()))
+	} else {
+		if template.Spec.Visibility == kubermaticv1.PolicyTemplateVisibilityProject {
+			if template.Spec.ProjectID == "" {
+				allErrs = append(allErrs, field.Required(specPath.Child("projectID"), fmt.Sprintf("projectID is required when visibility is '%s'", kubermaticv1.PolicyTemplateVisibilityProject)))
+			}
+			if template.Spec.Target != nil && template.Spec.Target.ProjectSelector != nil {
+				allErrs = append(allErrs, field.Forbidden(specPath.Child("target", "projectSelector"), fmt.Sprintf("projectSelector cannot be set when visibility is '%s'; scope is defined by projectID", kubermaticv1.PolicyTemplateVisibilityProject)))
+			}
+		}
+
+		if template.Spec.Visibility == kubermaticv1.PolicyTemplateVisibilityGlobal {
+			if template.Spec.ProjectID != "" {
+				allErrs = append(allErrs, field.Invalid(specPath.Child("projectID"), template.Spec.ProjectID, fmt.Sprintf("projectID must be empty when visibility is '%s'", kubermaticv1.PolicyTemplateVisibilityGlobal)))
+			}
+		}
+	}
```

Comment:
heh, I meant to use `switch` but we don't need the `else {}` block. Simply move out the `switch` from the `else {}` and naturally remove the empty `else {}`.
```suggestion
	}

	switch template.Spec.Visibility {
	case kubermaticv1.PolicyTemplateVisibilityProject:
		if template.Spec.ProjectID == "" {
			allErrs = append(allErrs, field.Required(specPath.Child("projectID"), fmt.Sprintf("projectID is required when visibility is '%s'", kubermaticv1.PolicyTemplateVisibilityProject)))
		}
		if template.Spec.Target != nil && template.Spec.Target.ProjectSelector != nil {
			allErrs = append(allErrs, field.Forbidden(specPath.Child("target", "projectSelector"), fmt.Sprintf("projectSelector cannot be set when visibility is '%s'; scope is defined by projectID", kubermaticv1.PolicyTemplateVisibilityProject)))
		}
	case kubermaticv1.PolicyTemplateVisibilityGlobal:
		if template.Spec.ProjectID != "" {
			allErrs = append(allErrs, field.Invalid(specPath.Child("projectID"), template.Spec.ProjectID, fmt.Sprintf("projectID must be empty when visibility is '%s'", kubermaticv1.PolicyTemplateVisibilityGlobal)))
		}
	}
```